ui/src/StateContainer.js

```diff
@@ -104,15 +117,12 @@ export class StateContainer extends LitElement {
     }
 
     /**
-     * Updates the isRecording and audioStream properties with the most up-to-date
+     * Updates the audioStream property with the most up-to-date
      * data from the vox-record-button component.
      */
-    handleUpdateWave() {
-        const recordComponent = this.viewShadowRoot.querySelector(
-            'vox-record-button'
-        );
-        this.isRecording = recordComponent.getIsRecording();
-        this.audioStream = recordComponent.getAudioStream();
+    handleUpdateWave(e) {
+        this.audioStream = e.detail.audioStream;
```

Comment:
It looks like you can remove this.audioStream from the state if you don't plan on doing anything with the audioStream.
It's being passed into the view component which is being passed into the recording view component which is passed into the button component.